examples/mnist.py

```diff
@@ -29,9 +29,9 @@ def main(save_to, num_epochs, bokeh=False):
               weights_init=IsotropicGaussian(0.01),
               biases_init=Constant(0))
     mlp.initialize()
-    x = tensor.matrix('features')
+    x = tensor.tensor4('features')
```

Comment:
There's an even simpler way to make sure inputs are flat: pass `flatten=('features',)` to the `MNIST` constructor.